src/Listener/SearchListener.php

```diff
@@ -0,0 +1,41 @@
+<?php
+namespace Crud\Listener;
+
+use Cake\Core\Plugin;
+use Cake\Event\Event;
+use RuntimeException;
+
+class SearchListener extends BaseListener
+{
+
+    /**
+     * Returns a list of all events that will fire in the controller during its lifecycle.
+     * You can override this function to add your own listener callbacks
+     *
+     * @return array
+     */
+    public function implementedEvents()
+    {
+        return [
+            'Crud.beforePaginate' => ['callable' => 'beforePaginate']
+        ];
+    }
+
+    /**
+     * Inject search conditions to the qeury object.
+     *
+     * @param \Cake\Event\Event $event
+     * @return void
+     */
+    public function beforePaginate(Event $event)
```

Comment:
So that both `beofrePaginate` and `beforeLookup` can be handled by the same method; and having configuration in the listener so that you can disable one of the events if needed